source/server/filter_chain_manager_impl.h

```diff
@@ -21,10 +28,71 @@ class FilterChainFactoryBuilder {
   buildFilterChain(const ::envoy::api::v2::listener::FilterChain& filter_chain) const PURE;
 };
 
+class FilterChainFactoryContextImpl : public Configuration::FilterChainFactoryContext,
```

Comment:
Can you add some comments above this explaining its relationship to filter chain, listener, etc, when it get created, destroyed, what it's used for?

Reply:
Added the life time